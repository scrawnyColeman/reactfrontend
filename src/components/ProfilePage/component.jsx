import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import PersonalInformation from '../PersonalInformation/component';
import UserCourses from '../UserCourses/component';
import { fetchUserLessons } from '../../data/apiCalls.js';
import BigButton from '../BigButton/component';
import { errorLogger } from '../../data/errorLogger';

const StyledItemWrapper = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr;
    margin: 20vh 20px 0;
    height: 60vh;
`;

const StyledButtonContainer = styled.div`
    width: 30vw;
    margin: auto;
`;

const ProfilePage = ({ setLocalUsername }) => {
    const user = {
        username: sessionStorage.getItem('activeUser'),
        email: sessionStorage.getItem('activeEmail'),
        id: sessionStorage.getItem('activeId'),
        userRole: sessionStorage.getItem('activeType'),
    };
    const [userLessons, setUserLessons] = useState(null);
    useEffect(() => {
        fetchUserLessons(user.id)
            .then((response) => {
                setUserLessons(response.data);
            })
            .catch(errorLogger);
    }, [user.id]);

    return userLessons ? (
        <StyledItemWrapper>
            <PersonalInformation user={user} setLocalUsername={() => setLocalUsername()} />
            {userLessons.length !== 0 ? (
                <UserCourses courses={userLessons} id={user.id} />
            ) : (
                <StyledButtonContainer>
                    <BigButton text={`Begin Learning`} path={`/learn`} />
                </StyledButtonContainer>
            )}
        </StyledItemWrapper>
    ) : (
        <div />
    );
};

export default ProfilePage;
