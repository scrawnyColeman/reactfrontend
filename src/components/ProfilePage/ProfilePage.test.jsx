import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import ProfilePage from './component';

Enzyme.configure({ adapter: new Adapter() });

describe('ProfilePage component', () => {
    test('component matches snapshot', () => {
        const wrapper = shallow(<ProfilePage />);
        expect(wrapper).toMatchSnapshot();
    });
});
