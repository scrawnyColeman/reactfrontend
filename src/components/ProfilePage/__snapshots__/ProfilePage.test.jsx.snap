// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage component component matches snapshot 1`] = `""`;
